usersession/userd/launcher.go

```diff
@@ -123,6 +130,157 @@ func (s *Launcher) OpenURL(addr string, sender dbus.Sender) *dbus.Error {
 	return nil
 }
 
+// OpenDesktopEntryEnv implements the 'OpenDesktopEntryEnv' method of the 'io.snapcraft.Launcher'
+// DBus interface. The desktop_file_id is described here:
+// https://standards.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html#desktop-file-id
+func (s *Launcher) OpenDesktopEntryEnv(desktop_file_id string, env []string, sender dbus.Sender) *dbus.Error {
+	desktop_file, err := desktopFileIdToFilename(existsOnFileSystem, desktop_file_id)
+	if err != nil {
+		return dbus.MakeFailedError(err)
+	}
+
+	exec_command, err := readExecCommandFromDesktopFile(desktop_file)
```

Comment:
I'm kind of surprised that we need to read and parse the desktop files ourselves and invoke the command therein, isn't there any utility or library that can do this for us?  We also end up being the parent process of these as @zyga mentions which means we need to keep track of them somehow.

Reply:
> I'm kind of surprised that we need to read and parse the desktop files ourselves and invoke the command therein, isn't there any utility or library that can do this for us?

I looked, but I couldn't find a Go library for this.

> We also end up being the parent process of these as @zyga mentions which means we need to keep track of them somehow.

No, we end up being the parent of `sh` which launches a process and exits.